tests/playwright-test/only-changed.spec.ts

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test as baseTest, expect, playwrightCtConfigText } from './playwright-test-fixtures';
+import { execSync } from 'node:child_process';
+
+const test = baseTest.extend({
+  setupRepository: async ({ writeFiles }, use, testInfo) => {
+    const baseDir = testInfo.outputPath();
+
+    const git = (command: string) => execSync(`git ${command}`, { cwd: baseDir });
+
+    await use(async () => {
+      await writeFiles({
```

Comment:
I'd move the `writeFiles` and `commit` calls outside of the fixture into each test. This will make it obvious in each test what's going on.

Reply:
done in https://github.com/microsoft/playwright/pull/31727/commits/16123cd9140f745d9c52b20163f120faa90dcfcb